### PR TITLE
Multiple colours within multi series via nth index

### PR DESCRIPTION
**Pull request by jdunkerley**
Using css nth index with the multi class to change the stroke and fill styles for each item.

### CODE REVIEW

File: src/series/series.css [hunk 1]
```diff
@@ -18,21 +18,47 @@
...
 }
 
-path.line {
-  fill: none;
+svg {
   stroke: #06c;
+  fill: #9cf;
 }
 
-path.area {
```
Comment (chrisprice) [inline src/series/series.css]: Just `text` maybe?

File: src/series/series.css [hunk 2]
```diff
@@ -18,21 +18,47 @@
...
-  fill: #9cf;
+svg text {
+  stroke: none;
+  fill: #000;
 }
 
-.point>circle {
```
Comment (chrisprice) [inline src/series/series.css]: This is exactly what I was thinking.

File: src/series/series.css [hunk 3]
```diff
@@ -18,21 +18,47 @@
   stroke: #6c0;
```
Comment (chrisprice) [inline src/series/series.css]: For all the ones above, could we drop the `>path`?

**Comment by chrisprice** [comment]
Looking good! 

Are you going to add some `*-opacity` styles for `area`, `line`, `point`, etc. or are you going to do that with default decorators? Maybe a good first stab would be to do it all in CSS?

File: src/series/series.css (hunk 3 above)
Comment (jdunkerley) [inline src/series/series.css]: yep thats a good change

**Comment by jdunkerley** [comment]
on this PR, I had put inline styles for line chart, will change to CSS.

I still wonder as to whether a class on all series so we could apply more targeted (e.g. make .series.area, .series.bar) could then .series.up and .series.down for the candlestick and ohlc colour but guess that is a different issue.

**Comment by chrisprice** [comment]
Sorry I missed that in the diff, I've been talked around to it being in the CSS :smile: 

I'm still to be convinced of `.series`, can you think of another usage of it?

As part of this, do you think it would make sense to centralise the CSS into one file?

**Comment by jdunkerley** [comment]
We have about 5 CSS files at the moment. Think it could be nice to have them in one place as it would make it easier to come up with a normalised pattern.

**Comment by jdunkerley** [comment]
Main thing I was looking to gain from the series class was not affecting the whole of svg. This could be done via the `cartesianChart` and `linearTimeSeries`, adjusting the CSS for their classes to set the default. That way it wouldn't affect all svg elements.

File: src/series/series.css [hunk 4]
```diff
@@ -1,38 +1,72 @@
 .candlestick>path {
```
Comment (chrisprice) [inline src/series/series.css]: Do we need the `>path` here?

File: src/indicator/indicator.css [hunk 5]
```diff
@@ -26,4 +26,10 @@
 
 .bollinger path.area {
```
Comment (chrisprice) [inline src/indicator/indicator.css]: Do we need the `.path`?

File: src/series/series.css (hunk 4 above)
Comment (jdunkerley) [inline src/series/series.css]: nope just blind as a bat

File: src/series/series.css [hunk 6]
```diff
@@ -1,38 +1,72 @@
...
+  fill: #f9f;
+}
+
+.area {
+  stroke-opacity: 0.0;
+}
+
+.line {
+  fill-opacity: 0.0;
+}
```
Comment (chrisprice) [inline src/series/series.css]: Newline required

File: src/indicator/indicator.css [hunk 7]
```diff
@@ -1,29 +1,30 @@
-.divergence .bar>path {
+.macd-divergence .bar {
```
Comment (chrisprice) [inline src/indicator/indicator.css]: I'd be tempted to drop all of these rules?

Comment (chrisprice) [inline src/indicator/indicator.css]: (for .macd-divergence .bar)

**Comment by chrisprice** [comment]
> We have about 5 CSS files at the moment. Think it could be nice to have them in one place as it would make it easier to come up with a normalised pattern and tidy up a little. For example, I think we have a chart class defined in fc.css not used anywhere.

:+1: Move it all into fc.css?

> Main thing I was looking to gain from the series class was not affecting the whole of svg. This could be done via the cartesianChart and linearTimeSeries, adjusting the CSS for their classes to set the default. That way it wouldn't affect all svg elements.

I don't think we need to worry about affecting the whole svg, the CSS should only be considered a starting point for anyone looking to use the library. In fact, maybe we should add some documentation to the top of the file stating as much?

**Comment by jdunkerley** [comment]
Spark lines are not rendering quite right at the moment. 
image

**Comment by jdunkerley** [comment]
Spark lines are not rendering correctly.

image

**Comment by ColinEberhardt** [comment]
I'm very happy with this change :-)

On thing I didn't spot in the diff, have you removed all the logic in the visual tests where colours are 'manually' applied to stacked / grouped series?

Once merged, this will also require some website update too ;-)

File: src/fc.css [hunk 8]
```diff
@@ -12,3 +12,214 @@
...
+.domain,
+.tick line,
+.tick path {
+  fill: none;
+  stroke-width: 1.0;
+  stroke: #bbb;
+}
+
+.tick text {
+  font: 10px sans-serif;
```
Comment (chrisprice) [inline src/fc.css]: Kill it

Comment (jdunkerley) [inline src/fc.css]: Removed, have likewise removed the one on annotations text

**Comment by jdunkerley** [comment]
Yes @ColinEberhardt have removed all the decorate stuff for stacked and grouped.

All the visual tests are working, still have a little more work on examples to do.

**Comment by jdunkerley** [comment]
I have gone over all the examples and documentation on the website and tweaked where needed (not very much).

A fix for the CrossHairs style so back to pale blue. 
All visualTests working (including another patch to crossHairs tests).

Think this is basically ready now.

File: site/src/examples/low-barrel/index.js [hunk 9]
```diff
@@ -298,7 +302,7 @@
             });
 
         var multi = fc.series.multi()
-            .series([gridlines, line, area, brush])
```
Comment (chrisprice) [inline site/src/examples/low-barrel/index.js]: Do we not need a line here?

Comment (jdunkerley) [inline site/src/examples/low-barrel/index.js]: used a decorate to reveal the stroke on the area. Seemed more efficient

Comment (chrisprice) [inline site/src/examples/low-barrel/index.js]: Oh cool, if it gives the same effect I've no problem with that

Comment (ColinEberhardt) [inline site/src/examples/low-barrel/index.js]: One thing to bear in mind is that a stroking an area will cause it to have an outline on all of its edges, whereas a combination of line & area gives the impression that only the 'top' edge of the area has been stroked. Although looking at the visuals I can't see that it makes any difference here.

**Comment by ColinEberhardt** [comment]
Great job - thanks.

**Pull request merged**